test/unit/marked-spec.js

```diff
@@ -1059,4 +1059,21 @@ br
     });
     expect(marked('*text*').trim()).toBe('<p><em>text walked</em></p>');
   });
+
+  it('should wait for async `walkTokens` function', async() => {
+    marked.use({
+      async: true,
+      async walkTokens(token) {
+        if (token.type === 'em') {
+          await new Promise((resolve) => {
+            setTimeout(resolve, 100);
+          });
+          token.text += ' walked';
+          token.tokens = this.Lexer.lexInline(token.text);
+        }
+      }
+    });
+    const html = await marked('*text*');
```

Comment:
Since `await` also works with anything, it might be good to verify this actually returns a promise

```
const p = marked('*text*')
expect(p instanceof Promise)
cosnt html = await p
```